CRM/Contribute/BAO/Contribution.php

```diff
@@ -141,14 +140,19 @@ public static function add(&$params, $ids = []) {
         $params['creditnote_id'] = self::createCreditNoteId();
       }
     }
-    if (empty($params['contribution_status_id'])) {
+    $contributionStatusID = $params['contribution_status_id'] ?? NULL;
```

Comment:
Forgive my ignorance, but I don't see what conditions would lead the '?? NULL' to change the value of $contributionStatusID. Whatever is in $params['contribution_status_id'] will be assigned, whether it is NULL or not, regardless of the Null coalescing operator, no?

Reply:
it gets populated down on line 155

Reply:
So it will be set to NULL if for some reason `$params['contribution_status_id']` is not set and seems sensible here